src/components/canrawview/canrawview_p.h

```diff
@@ -90,17 +121,41 @@ class CanRawViewPrivate : public QWidget {
     CanRawView* q_ptr;
     int prevIndex = 0;
     int rowID = 0;
+    QStringList columnsOrder;
+
+    void writeColumnsOrder(QJsonObject& json) const
+    {
+        uint ii = 0;
+        QJsonArray columnList;
+        for (auto iter : columnsOrder) {
+            if (ui->tv->isColumnHidden(ii++) == false) {
+                columnList.append(iter);
+            }
+        }
+        json["Columns"] = columnList;
+    }
+
+    void writeViewModel(QJsonArray& jsonArray) const
+    {
+        for (auto ii = 0; ii < tvModel.rowCount(); ++ii) {
```

Comment:
`int rowCount() const` (is that this member function?) I would put `int row = 0;`.